### ISSUE TITLE
Experiment results

### ISSUE THREAD

**Issue by PeterHJY628** [luiscarlosgph/jiayuan-huang-phd#28]
# New Experiment: 
DINO V3 + single linear classfication layer: 

## result
Initial query || Query Size: 1per class (10 in total)  || 10
Budget: 200
[report_single_file.html](https://github.com/user-attachments/files/26613468/report_single_file.html)
Transition Point：
- Least Confidence: ~30 (Iteration 3)

<img width="1087" height="623" alt="Image" src="https://github.com/user-attachments/assets/bc4267a0-c9df-41df-ac7e-ae1223189d8a" />

- Entropy: ~30 (Iteration 3)

- Badge: No cold satart.
<img width="1087" height="623" alt="Image" src="https://github.com/user-attachments/assets/4789df03-ec84-4bd8-a31f-f5f662be51fd" />

<img width="1087" height="623" alt="Image" src="https://github.com/user-attachments/assets/5e765358-5b1a-428c-beb0-bb916c891988" />

- Coreset: better than random at first, then significantly worse than it 

<img width="1087" height="623" alt="Image" src="https://github.com/user-attachments/assets/3ffd55cc-4eb1-4ede-9c3a-c6c55017ffc1" />

Initial query || Query Size: 2per class (20 in total)  || 20
Budget: 200
[report_single_file.html](https://github.com/user-attachments/files/26613505/report_single_file.html)
Transition Point：
- Least Confidence: No cold start shown since the initial size is 20, under this condition, the classification head is well trained

<img width="1087" height="623" alt="Image" src="https://github.com/user-attachments/assets/121fca81-2b8a-4e6b-a4fb-4fcad20069db" />

- Entropy: ~40 (Iteration 2)

<img width="1087" height="623" alt="Image" src="https://github.com/user-attachments/assets/a238590b-0c9a-4a46-a85c-b672a1cdf59b" />

- Badge: No cold satart.

<img width="1087" height="623" alt="Image" src="https://github.com/user-attachments/assets/4d04ce05-4d11-4d4a-9257-15f0c547c956" />

- Coreset: better than random at first, then significantly worse than it 

<img width="1087" height="623" alt="Image" src="https://github.com/user-attachments/assets/c0de1f52-c71c-47aa-a1af-c3dc654525fb" />

Initial query || Query Size: 3per class (30 in total)  || 30
Budget: 200
Transition Point：
- Least Confidenc: No transition point, always better

<img width="1087" height="623" alt="Image" src="https://github.com/user-attachments/assets/282142ea-2bb7-4ccc-9f65-acfb2e3333ea" />

- Entropy:  60 (Iteration 2)

<img width="1087" height="623" alt="Image" src="https://github.com/user-attachments/assets/8bb7eaa2-8026-42d8-9ebd-c17f5074aff5" />

- Badge: No cold satart.

<img width="1087" height="623" alt="Image" src="https://github.com/user-attachments/assets/f97d6533-3f55-4dda-9aa6-9eba585bb41a" />

- Coreset: better than random at first, then significantly worse than it 

<img width="1087" height="623" alt="Image" src="https://github.com/user-attachments/assets/f6d7973c-07ac-4f41-a070-83ed3e821e08" />

[report_single_file.html](https://github.com/user-attachments/files/26613515/report_single_file.html)

## Finding
1. The cold start stage of the pretrained model is very short. It is probably due to the noise of classfication head. It proves that a well-trained representation helps the uncertainty.
2.  The diversity based method performs significantly  worse than random sampling.

## Insights
1. we need to run this network-architecture on other datasets to verify if this phenomenon is unified?
Experiments are finished, I use the same network to run it on CIFAR100, the result is:
Initial query || Query Size: 1per class (10 in total)  || 10
Budget: 200
Transition Point：
- Least Confidenc:  After a shor cold start, the LC outperforms Random, but is surpassed by random at budget point 180

<img width="1087" height="622" alt="Image" src="https://github.com/user-attachments/assets/8592b66c-1425-4ae4-a15c-816a3c2529af" />

- Entropy: No cold start at first, but will be surpassed by random at budget point 170

<img width="1087" height="622" alt="Image" src="https://github.com/user-attachments/assets/5464fb45-848f-499d-9d81-0a04f47d03bd" />

- Badge: always better than random

<img width="1087" height="623" alt="Image" src="https://github.com/user-attachments/assets/0395f2f9-4534-4831-bb62-2d0af787a214" />

- Corset: always better than random

<img width="1087" height="622" alt="Image" src="https://github.com/user-attachments/assets/061587f4-f19b-4f34-980d-de8f802910ea" />

It is very suprings that the corset on different datasets has conpletely reverse results. The comparision is shown below:
Corset result on cifar10
<img width="1087" height="623" alt="Image" src="https://github.com/user-attachments/assets/3ffd55cc-4eb1-4ede-9c3a-c6c55017ffc1" />

on Cifar 100
<img width="1087" height="622" alt="Image" src="https://github.com/user-attachments/assets/061587f4-f19b-4f34-980d-de8f802910ea" />
### Analyzation
Since CIFAR-10 has only 10 classes, random sampling can better preserve the original data distribution, making it more likely to select well-represented samples near the cluster centers. In contrast, Coreset, in its pursuit of maximizing feature space coverage, may end up sampling outliers or noise.

On the other hand, for CIFAR-100, due to the low budget, the random sampling method might fail to cover all classes or result in highly imbalanced sample sizes across classes. Coreset, however, manages to encompass samples from all categories during its process of maximizing feature space coverage."

It proves that the coverage of sample space is very important once again
## Discussion
I have running the cold start experiments for 3 weeks. In this progress, I made lots of mistakes, inclusing
1. At first, I didn't detemine the seed of GPU, which leads to diffrent initial results when using the same seed. so the result obtained in the first week is not reproduciable
2. I modified the code base provided by the paper "cross-domain active learning benchmark ".
3. Although I freeze the backbone of DINO v3, It takes one week to get the result of one experiment setting of only two methods. Then I realized that the forward pass of DINO v3 took lots of time, So I saved the embedding after forward passing so that it can be used in the following training progress. It boots up the training now.

**Comment by PeterHJY628** [comment]
# Four phases of our experiments:
## Phase 1: Run baselines, stockpile checkpoints

CIFAR-10 + randomly initialised ResNet-18. Run only Random and Entropy sampling across multiple seeds. Save model weights and labeled-set indices after every AL round. Goal: obtain reproducible cold-start behaviour.
status: done

## Phase 2: Plot curves, locate the transition band

No GPU needed — analyse Phase 1 CSV logs locally. Plot Accuracy/Loss curves with error bands, find where Entropy consistently overtakes Random, and mark three target rounds: $t_{early}$ (AL underperforms), $t_{trans}$ (crossover), $t_{late}$ (AL dominates).

status: done

## Phase 3: Add pretrained backbone, run ablation

Same setup but with a frozen pretrained ResNet-18 (SimCLR weights), training only the linear head. Compare against Phase 1 scratch group: if the pretrained group shows near-zero cold start, it confirms poor representation quality — not insufficient labels — as the root cause.

stauts: running

## Phase 4: Compute metrics, explain the mechanism

Load Phase 1/3 checkpoints, extract penultimate-layer features at each round, compute CKA similarity and Wasserstein distance between scratch and pretrained representations. If these metrics shift sharply around $t_{trans}$, it provides a representation-level explanation for why the crossover happens when it does.

status: to do.
